### ISSUE TITLE
[BUG] `quicksave` latches onto first savegame since executable launch. Not even `endgame` can change the `quicksave` command's target file, game must be restarted to pick a new quicksave.

### ISSUE THREAD

**Issue by ipaqmaster** [ZDoom/gzdoom#2118]
### GZDoom version

GZDoom g4.11pre-254-gf4dc16b27-m - 2023-08-03 13:03:41 -0400 - SDL version

### Which game are you running with GZDoom?

Doom 2

### What Operating System are you using?

Linux x86_64

### Please describe your specific OS version

Archlinux. Kernel version 6.4.8-arch1-1

### Relevant hardware info

3900x, 2080Ti (Windforce), CT2000 NVMe, 32G DDR4@3600MT/s

### Have you checked that no other similar issue already exists?

- [X] I have searched and not found similar issues.

### A clear and concise description of what the bug is.

`quicksave` and `quickload` together latch onto the first savegame you specify since the game's execution. Up to this point this behavior is normal and expected. But the game makes no attempt to dereference the save file once the "game" is over, loaded nor when "New Game" is called.

In modern times, I've gotten into a habit of making a new 'forked' savegame via the main menu in various situations such as when I get to a new level, when I want to test different ideas or run attempts. Unfortunately no matter what I try, the `quicksave` and `quickload` filename variable cannot be influenced via the regular game menu experience. I also tried `endgame` however that doesn't reset the context either.

I feel expect certain gamestate changes such as Loading a different save filename from the menu, hitting "New Game" or issuing `endgame` to reset this variable without having to dive into get/set commands manually.

### Steps to reproduce the behaviour.

Explain how to reproduce
1. Use `quicksave` (Or a bind to it)
2. Write a new savefile or pick an existing one for `quicksave`.
3. As expected - invoking `quicksave` or `quickload` will now refer to the file initially created/selected.
4. This value is 'permanent' per launch. There is no way to reset this variable without restarting gzdoom.

No amount of selecting "New Game", "Load Game" or issuing `endgame` (Or a bind to it) resets the targeted savegame file which `quicksave` and `quickload` refer to.

### Your configuration

```ini
NA

Reproducible with `gzdoom -config /tmp/gzdoomtest.cfg -iwad .config/gzdoom/DOOM2.WAD -savedir /tmp` simulating a fresh installation.
```

### Provide a Log

Can provide on request - the report is in relation to behavior of the software so I'm hoping this is NA.

**Comment by RicardoLuis0** [comment]
isn't that just how quicksave works by default? if you want it to use unique files, enabling quicksave rotation would do it

**Comment by ipaqmaster** [comment]
I saw that while digging through the source last night and tried it - I'm not so excited about quicksave rotation is that it rotates over the save files after x rotations. So I'm either overwriting the past in the end (which is exactly what I wish quicksave could be told not to do after the player isn't using that save anymore) or setting the rotation count variable to a ridiculously high rotation number and flooding my savedir.

It would be so amazing if quicksave could just take the hint and reset itself when the save file it was writing to is no longer what's being run... if not some flag somewhere to enable that behaviour.


I poked around at the code last night - the CCMD and actual "do save" functions don't seem to expose any variables for easy resetting via the console. Bringing this idea to fruition would require tweaks in other CCMD calls to reset it after certain gamestate changes if a user desires (perhaps a new default-off options variable to toggle this idea on?)
